packages/components/src/tooltip/test/index.js

```diff
@@ -169,79 +170,77 @@ describe( 'Tooltip', () => {
 				</Tooltip>
 			);
 
-			const button = wrapper.find( 'button' );
-			button.simulate( 'mouseenter', { type: 'mouseenter' } );
-
-			const popoverBeforeTimeout = wrapper.find( 'Popover' );
-			expect( popoverBeforeTimeout ).toHaveLength( 0 );
+			const button = screen.getByRole( 'button' );
+			await user.hover( button );
+			expect( screen.queryByText( 'Help text' ) ).not.toBeInTheDocument();
 			expect( originalMouseEnter ).toHaveBeenCalledTimes( 1 );
 
 			// Popover does not yet exist after default delay, because custom delay is passed.
 			setTimeout( () => {
-				const popoverBetweenTimeout = wrapper.find( 'Popover' );
-				expect( popoverBetweenTimeout ).toHaveLength( 0 );
+				expect(
+					screen.queryByText( 'Help text' )
+				).not.toBeInTheDocument();
 			}, TOOLTIP_DELAY );
-
 			// Popover appears after custom delay.
 			setTimeout( () => {
-				const popoverAfterTimeout = wrapper.find( 'Popover' );
-				expect( popoverAfterTimeout ).toHaveLength( 1 );
-
+				expect( screen.getByText( 'Help text' ) ).toBeInTheDocument();
 				jest.runOnlyPendingTimers();
 				jest.useRealTimers();
 			}, 2000 );
 		} );
 
-		it( 'should show tooltip when an element is disabled', () => {
-			const wrapper = mount(
+		it( 'should show tooltip when an element is disabled', async () => {
```

Comment:
Let's not use `fakeTimers` and `setTimeout`

```diff
diff --git a/packages/components/src/tooltip/test/index.js b/packages/components/src/tooltip/test/index.js
index 1057d6eee1..c36174dce9 100644
--- a/packages/components/src/tooltip/test/index.js
+++ b/packages/components/src/tooltip/test/index.js
@@ -178,18 +178,30 @@ describe( 'Tooltip', () => {
 				</Tooltip>
 			);
 
-			const button = screen.getByRole( 'button' );
+			const button = screen.getByRole( 'button', { name: 'Click me' } );
+			expect( button ).toBeInTheDocument();
+			expect( button ).toBeDisabled();
+
+			// Note: this is testing for implementation details,
+			// but couldn't find a better way.
 			const buttonRect = button.getBoundingClientRect();
-			const eventCatcher =
-				container.getElementsByClassName( 'event-catcher' )[ 0 ];
+			const eventCatcher = container.querySelector( '.event-catcher' );
 			const eventCatcherRect = eventCatcher.getBoundingClientRect();
 			expect( buttonRect ).toEqual( eventCatcherRect );
 
 			await user.hover( eventCatcher );
 
-			setTimeout( () => {
-				expect( screen.getByText( 'Help text' ) ).toBeInTheDocument();
-			}, TOOLTIP_DELAY );
+			// Tooltip hasn't appeared yet
+			expect(
+				screen.queryByText( 'Show helpful text here' )
+			).not.toBeInTheDocument();
+
+			act( () => jest.advanceTimersByTime( TOOLTIP_DELAY ) );
+
+			// Tooltip shows after the delay
+			expect(
+				screen.getByText( 'Show helpful text here' )
+			).toBeInTheDocument();
 		} );
 
 		it( 'should not emit events back to children when they are disabled', async () => {
```


(also, I don't like how this test is still very coupled to the implementation details, but I'm not sure there's a better way — cc @mirka )

Reply:
Fixed by 7f0367a